app/models/batch_context.rb

```diff
@@ -108,55 +108,55 @@ def normalize_dir(dir)
     dir&.chomp('/')
   end
 
-  def normalize_bundle_dir
-    self[:bundle_dir] = normalize_dir(bundle_dir)
+  def normalize_staging_location
+    self[:staging_location] = normalize_dir(staging_location)
   end
 
-  def bundle_dir_path
-    return unless bundle_dir
+  def staging_location_path
+    return unless staging_location
 
-    Pathname.new(bundle_dir).expand_path
+    Pathname.new(staging_location).expand_path
   end
 
   def output_dir_exists!
     return if Dir.exist?(output_dir)
 
-    errors.add(:bundle_dir, "Output directory (#{output_dir}) should already exist, but doesn't")
+    errors.add(:staging_location, "Output directory (#{output_dir}) should already exist, but doesn't")
     throw(:abort)
   end
 
   def output_dir_no_exists!
     if Dir.exist?(output_dir)
-      errors.add(:bundle_dir, "Output directory (#{output_dir}) should not already exist")
+      errors.add(:staging_location, "Output directory (#{output_dir}) should not already exist")
       throw(:abort)
     end
     FileUtils.mkdir_p(output_dir)
   end
 
   # rubocop:disable Metrics/AbcSize
-  def verify_bundle_directory
-    return if errors.key?(:bundle_dir)
-    return errors.add(:bundle_dir, "'#{bundle_dir}' not found.") unless File.directory?(bundle_dir)
+  def verify_staging_location
+    return if errors.key?(:staging_location)
+    return errors.add(:staging_location, "'#{staging_location}' not found.") unless File.directory?(staging_location)
 
-    errors.add(:bundle_dir, "missing manifest: #{bundle_dir}/#{manifest}") unless File.exist?(File.join(bundle_dir, manifest))
-    errors.add(:bundle_dir, "missing file manifest: #{bundle_dir}/#{file_manifest}") if using_file_manifest && !File.exist?(File.join(bundle_dir, file_manifest))
+    errors.add(:staging_location, "missing manifest: #{staging_location}/#{manifest}") unless File.exist?(File.join(staging_location, manifest))
+    errors.add(:staging_location, "missing file manifest: #{staging_location}/#{file_manifest}") if using_file_manifest && !File.exist?(File.join(staging_location, file_manifest))
   end
   # rubocop:enable Metrics/AbcSize
 
-  def verify_bundle_dir_path
-    return if errors.key?(:bundle_dir)
+  def verify_staging_location_path
+    return if errors.key?(:staging_location)
 
     match_flag = nil
-    bundle_dir_path&.ascend do |sub_path|
-      next unless ::ALLOWABLE_BUNDLE_DIRS.include?(sub_path.to_s)
+    staging_location_path&.ascend do |sub_path|
+      next unless ::ALLOWABLE_STAGING_LOCATIONS.include?(sub_path.to_s)
 
       match_flag = sub_path
       break
     end
-    # match_flag = nil means we are not in the sub path, match_flag == bundle_dir_path means the user only entered the root path.
-    return unless match_flag.nil? || match_flag == bundle_dir_path
+    # match_flag = nil means we are not in the sub path, match_flag == staging_location means the user only entered the root path.
+    return unless match_flag.nil? || match_flag == staging_location
```

Comment:
I think this is supposed to be `staging_location_path` ...

Reply:
fixed